python/sparkdl/graph/input.py

```diff
@@ -0,0 +1,317 @@
+        #                any signature_def fields in a checkpoint's meta_graph_def.
+        meta_graph_def = meta_graph_pb2.MetaGraphDef()
+        with open("{}.meta".format(ckpt_path), 'rb') as fin:
+            meta_graph_def.ParseFromString(fin.read())
+
+        saver = tf.train.import_meta_graph(meta_graph_def, clear_devices=True)
+        saver.restore(sess, ckpt_path)
+
+        if signature_def_key is not None:
+            sig_def = meta_graph_def.signature_def[signature_def_key]
+            gin = _build_with_sig_def(sess=sess, graph=graph, sig_def=sig_def)
+        else:
+            gin = _build_with_feeds_fetches(sess=sess, graph=graph, feed_names=feed_names,
+                                            fetch_names=fetch_names)
+    return gin
+
+
+def _from_saved_model_impl(saved_model_dir, tag_set, signature_def_key, feed_names, fetch_names):
+    """
+    Construct a TFInputGraph from a SavedModel.
+    Notice that one should either provide the `signature_def_key` or provide both
+    `feed_names` and `fetch_names`. Please set the unprovided values to None.
+
+    :param signature_def_key: str, name of the mapping contained inside the `signature_def`
+                              from which we retrieve the signature key to tensor names mapping.
+    :param feed_names: list, names of the input tensors.
+    :param fetch_names: list, names of the output tensors.
+    """
+    assert (feed_names is None) == (fetch_names is None), \
+        'feed_names and fetch_names, if provided must appear together'
+    assert (feed_names is None) != (signature_def_key is None), \
+        'must either provide feed_names or singnature_def_key'
+
+    graph = tf.Graph()
+    with tf.Session(graph=graph) as sess:
+        tag_sets = tag_set.split(',')
+        meta_graph_def = tf.saved_model.loader.load(sess, tag_sets, saved_model_dir)
+
+        if signature_def_key is not None:
+            sig_def = tf.contrib.saved_model.get_signature_def_by_key(meta_graph_def,
+                                                                      signature_def_key)
+            gin = _build_with_sig_def(sess=sess, graph=graph, sig_def=sig_def)
+        else:
+            gin = _build_with_feeds_fetches(sess=sess, graph=graph, feed_names=feed_names,
+                                            fetch_names=fetch_names)
+    return gin
+
+
+def _build_with_sig_def(sess, graph, sig_def):
+    # pylint: disable=protected-access,attribute-defined-outside-init
+    assert sig_def, \
+        'signature_def {} provided, '.format(sig_def) + \
+        'but failed to find it from the meta_graph_def'
+
+    with sess.as_default(), graph.as_default():
+        feed_mapping = {}
+        feed_names = []
+        for sigdef_key, tnsr_info in sig_def.inputs.items():
+            tnsr_name = tnsr_info.name
+            feed_mapping[sigdef_key] = tnsr_name
+            feed_names.append(tnsr_name)
+
+        fetch_mapping = {}
```

Comment:
we should add some tests that specifically check that these mappings are created correctly.